### PR TITLE
feat: introduce notification types

### PR DESCRIPTION
**Pull request by alexcarpenter**
## Description

Based on whether or not the notification contains interactive content, render the notification with `role="alert"` for toasts and `role="alerDialog"` for dialogs.

This is based off the work HDS team has done with their [toast implementations](https://design-system-components-hashicorp.vercel.app/components/toast).

And confirmation from HDS team https://hashicorp.slack.com/archives/C7KTUHNUS/p1657818549189699?thread_ts=1657652997.846709&cid=C7KTUHNUS

### Usage

```tsx
notification() // defaults to toast type
notification.toast() // type = toast with role = alert
notification.dialog() // type = dialog with role = alertDialog
```

https://user-images.githubusercontent.com/825855/179763363-611d8b2c-0f60-4864-b44e-f0dff14a8951.mp4

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

**Comment by alexcarpenter** [comment]
In use: https://github.com/hashicorp/hashicorp-www-next/pull/2923

File: packages/notification/components/notifications/index.tsx [hunk 1]
```diff
@@ -39,7 +45,7 @@ export const Notifications = ({
                 scale: reducedMotion ? 1 : 0.75,
                 transition: { duration: 0.2 },
               }}
-              role="status"
+              role={role}
```
Comment (ashleemboyer) [inline packages/notification/components/notifications/index.tsx:48]: (Question 🙋🏻) Is it possible to auto-detect the role based on whether or not the notification contains interactive content?

Comment (alexcarpenter) [reply packages/notification/components/notifications/index.tsx:48]: The thinking here was following the same reason we have explicit [Button](https://github.com/hashicorp/dev-portal/tree/main/src/components/button) and [ButtonLink](https://github.com/hashicorp/dev-portal/tree/main/src/components/button-link) components vs a single Button component that changes based on the attributes provided.

File: packages/notification/types.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ export type Renderable = JSX.Element | string | null
 
 export type ValueFunction<TValue, TArg> = (arg: TArg) => TValue
 export type ValueOrFunction<TValue, TArg> = TValue | ValueFunction<TValue, TArg>
+export type NotificationAsType = 'toast' | 'dialog'
```
Comment (dstaley) [inline packages/notification/types.ts:7]: nit: I think this type name would be better as `NotificationType`.

**Review by dstaley — APPROVED** [review]

File: packages/notification/types.ts (hunk 2 above)
Comment (alexcarpenter) [reply packages/notification/types.ts]: Updated https://github.com/hashicorp/react-components/pull/656/commits/a7e73d52f83d5a43f7989df4d832110033e15e75

**Pull request merged**